woocommerce/woocommerce-ios
# Product Add/Edit: Update price row and screen to support subscription

Part of #11135 


## What

This PR adds the support for updating the subscription period for simple subscription products. This setting is accessible through the price setting screen as discussed in p1699462947949809-slack-C03L1NF1EA3.

Due to the size of the changes, the subscription sign-up fee and more unit tests will be added in a separate PR.

## How
- Added a new row for the subscription period on the product price setting screen.
- Added a new extension for String and `ProductFormDataModel` to format subscription period. The format follows the read-only version instead of Core for simplicity.
- Created a separate use case for the picker view for subscription period and interval.
- Updated the `TitleAndTextFieldTableViewCell` to accept custom input view.
- Updated the subscription period row to display the new picker view for inputting the subscription period and interval.
- Updated the completion handler for price setting to include subscription period and interval.

## Testing instructions

- Enable the feature flag `subscriptionProducts` to be able to edit subscription products.
- Create a simple subscription product for your test site on wp-admin if you haven't already.
- Build the app, log in to your test store and navigate to the Products tab.
- Select a simple subscription product.
- Notice that the price row is displayed in place of the read-only subscription row. It should state the subscription period, like "$10 every 2 weeks".
- Tap the price row, notice a few changes:
  - A new row for subscription period is displayed below the Regular Price row. 
  - The Sale Price row is displayed in the second section with the new name Sale.
  - Tap on the Subscription Period row, notice that a picker is displayed for you to select the interval and period. 
  - Tap on Done on the toolbar above the picker, the new period should be displayed correctly.
  - Tap on Done on the price settings screen and tap Save on the product form, the new period should be updated correctly.

## Screenshots


<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/4699750e-041b-4b60-90f3-9edea1f02a94" width=320 />



---
